catroid/src/test/java/org/catrobat/catroid/test/content/actions/IfOnEdgeBounceActionTest.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions
+
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.SingleSprite
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.test.MockUtil
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class IfOnEdgeBounceActionTest(
+    private val name: String,
+    private val initialPosX: Float,
+    private val initialPosY: Float,
+    private val expectedPosX: Float,
+    private val expectedPosY: Float,
+    private val initialDirection: Int,
+    private val expectedDirection: Float
+) {
+
+    private lateinit var sprite: Sprite
+    private lateinit var project: Project
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0} {5}")
+        fun parameters() = (-179..180).flatMap { listOf(
```

Comment:
For readability reasons please replace `it` with `initialDirection`
`fun parameters() = (-179..180).flatMap { initialDirection -> listOf(`